app/javascript/src/Users/components/PermissionsForm.jsx

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import 'raf/polyfill'
+import 'core-js/es6/map'
+import 'core-js/es6/set'
+import 'core-js/es6/array'
+
+import React from 'react'
+
+import { Inputs } from 'Users/components/Inputs'
+import { RoleInput } from 'Users/components/RoleInput'
+import { UserRole } from 'Users/components/UserRole'
+import { FeatureAccessInput } from 'Users/components/FeatureAccessInput'
+import { FeatureAccess } from 'Users/components/FeatureAccess'
+import { ServiceFeatureAccess } from 'Users/components/ServiceFeatureAccess'
+import { ServiceAccessList } from 'Users/components/ServiceAccessList'
+import { ServiceAccess } from 'Users/components/ServiceAccess'
+
+import { isServicePermissionsGranted, getFeatureName, toggleElementInCollection } from 'Users/utils'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import type { Role, Feature } from 'Users/types'
+import type { Service } from 'Types'
+
+type Props = {
+  initialState: any,
+  features: Feature[],
+  services: Service[]
+}
+
+type State = {
+  role: Role,
+  checkedFeatures: Array<Feature | 'services'>,
+  checkedServicesIds: number[]
+}
+
+class PermissionsForm extends React.Component<Props, State> {
+  state = {
+    role: this.props.initialState.role || 'admin',
+    checkedFeatures: this.props.initialState.admin_sections || [],
+    checkedServicesIds: this.props.initialState.member_permission_service_ids || []
+  }
+
+  handleRoleChange = (role: Role) => this.setState({ role })
+
+  handleFeatureChecked = (feature: Feature | 'services') => {
```

Comment:
Is the string `services` a type `Feature` ?

Reply:
Clarification for this above https://github.com/3scale/porta/pull/604#discussion_r271355724